### ISSUE TITLE
Modify all SecurityGroups tagged with `kubernetes.io/cluster/$name`

### ISSUE THREAD

**Issue by ellistarn** [kubernetes-sigs/aws-load-balancer-controller#2367]
**Is your feature request related to a problem?**
Currently, the controller fails if there are multiple security groups tagged with `kubernetes.io/cluster/$name`.

**Describe the solution you'd like**
It's common to tag both controlplane and dataplane resources with `kubernetes.io/cluster/$name`. This is a powerful discovery mechanism to simplify the end user configuration burden. Right now, the lb controller assumes that there is only one security group with this tag, and that it's owned by the control plane. This creates compatibility issues like https://github.com/weaveworks/eksctl/issues/4054.

**Describe alternatives you've considered**
We could potentially use a different tag for the control plane security groups, but this may be challenging to coordinate with all providers (kops, capi, eks, etc)

**Comment by ellistarn** [comment]
@M00nF1sh any progress on prioritization here?

**Comment by armujahid** [comment]
I am also getting this issue if we use default karpenter provisioner in a newly created eks cluster https://karpenter.sh/v0.7.1/getting-started/getting-started-with-eksctl/#provisioner

**Comment by ellistarn** [comment]
@armujahid, we made karpenter flexible to arbitrary tags, so you should be able to work around this.

**Comment by armujahid** [comment]
@ellistarn yes that's what I have done by using another nodegroup selector like kubernetes.io/cluster/${CLUSTER_NAME}

**Comment by mkotsalainen** [comment]
@armujahid @ellistarn Can you tell me what security group selector that I can use to fix this? I also have a plain vanilla eksctl cluster.

**Comment by armujahid** [comment]
@mkotsalainen i am using  

kubernetes.io/cluster/${CLUSTER_NAME}: owned

Security group selector for my vanilla karpenter cluster

**Comment by mkotsalainen** [comment]
Ok thanks. Before I saw your answer I tried 
```
provider:
    securityGroupSelector:
      Name: "*eks-cluster-sg-${CLUSTER_NAME}*"`
```
and that seems to fix my problem (getting traefik ingress controller to work with karpenter)

**Comment by ellistarn** [comment]
@M00nF1sh any word on this one?

**Comment by hitsub2** [comment]
It does not get fixed, right?

**Comment by carflo** [comment]
My clusters ran into the same issue when we migrated over to Karpenter for node management (multiple SGs tagged with the cluster tag). 

I opened #3147 to allow users to specify additional tags to further specify the security group that the load balancer controller should use. My clusters are running with the changes in this PR and after specifying additional tags, this seems to be running well. Hoping my PR can get some feedback and make it to a release soon as this seems to be affecting several folks 🙏

**Comment by carflo** [comment]
@kishorj can you reopen this? I have a [PR](https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3147) that addresses this issue but I am awaiting feedback.

**Comment by ricardorqr** [comment]
Hello guys. I read all the comments and I'm still lost in this problem. [Karpenter official docs](https://karpenter.sh/docs/concepts/node-templates/#specsecuritygroupselector) has a note about this but I didn't bet how to solve it. Can anyone please give me directions?

**Comment by kitos9112** [comment]
@ricardorqr There's very little you can do unless your use case is simple enough (like mine) and you can renounce from tagging all other EKS-related components with the tag in question.

This might not work for everybody, but I have a dedicated Karpenter `Provisioner` CR with interpolated values from the Terraform stack (e.g., subnetIds, securityGroupIDs, etc...). That means I disabled the auto-discovery mechanism altogether.

Not ideal, but all you can do is use a single SG managed by the AWS LB controller for your EKS worker nodes. 

I hope https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3147 gets merged soon!

**Comment by ellistarn** [comment]
Has anyone had a chance to try out #3137? Would love a user acceptance test so that we can close this one out.

**Comment by kasvith** [comment]
I also hit the same problem with this, was scratching my head around why nothing works and find out its missing a SG.
Please solve this issue as others can go through same.

I also used Karpenter with eksctl #3420

**Comment by kitos9112** [comment]
The PR associated to this issue appears as complete. Is there an ETA for version 2.7?

**Comment by oliviassss** [comment]
@kitos9112, hi we are targeting on Dec 2023 for v2.7.0

**Comment by ldiniz-atacana** [comment]
Hi @oliviassss,
Any update on this?

**Comment by oliviassss** [comment]
Fix shipped with v2.7.0 last week
https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3147

**Comment by oliviassss** [comment]
The flag to config `ServiceTargetENISGTags` has been added in the helm chart v1.7.1 with the LBC v2.7.1 release, we should be good to close the tickets.
https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/tag/v2.7.1
